pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -63,14 +63,21 @@ func NewProvisioner(context *clusterd.Context, clusterInfo *client.ClusterInfo)
 	return &Provisioner{context: context, clusterInfo: clusterInfo}
 }
 
+// GenerateUserID implements Provisioner.GenerateUserID()
 func (p Provisioner) GenerateUserID(obc *v1alpha1.ObjectBucketClaim, ob *v1alpha1.ObjectBucket) (string, error) {
```

Comment:
IMO order should be,
fetch OB.
if OB is not nil, then check conflicts with old user and throws error
otherwise either returns user-specified  setting or generated the user name